src/Components/Server/src/Microsoft.AspNetCore.Components.Server.csproj

```diff
@@ -104,17 +104,14 @@
     <BlazorServerJSFile Condition=" '$(Configuration)' != 'Debug' ">..\..\Web.JS\dist\Release\$(BlazorServerJSFilename)</BlazorServerJSFile>
   </PropertyGroup>
 
-  <!-- blazor.server.js should exist after Microsoft.AspNetCore.Components.Web.JS.npmproj builds. Fall back if not. -->
-  <Target Name="_CheckBlazorServerJSPath" AfterTargets="ResolveProjectReferences" Condition=" !EXISTS('$(BlazorServerJSFile)') ">
-    <Warning Text="'$(BlazorServerJSFile)' does not exist. Falling back to checked-in copy." />
-    <PropertyGroup>
-      <BlazorServerJSFile>..\..\Web.JS\dist\Release\$(BlazorServerJSFilename)</BlazorServerJSFile>
-    </PropertyGroup>
+  <!-- blazor.server.js should exist after Microsoft.AspNetCore.Components.Web.JS.npmproj builds. -->
+  <Target Name="_CheckBlazorServerJSPath" AfterTargets="ResolveProjectReferences" Condition=" '$(BuildNodeJS)' != 'false' and !EXISTS('$(BlazorServerJSFile)') ">
+    <Error Text="'$(BlazorServerJSFile)' does not exist. Falling back to checked-in copy." />
   </Target>
 
   <Target Name="_AddEmbeddedBlazor" AfterTargets="_CheckBlazorServerJSPath">
     <ItemGroup>
-      <EmbeddedResource Include="$(BlazorServerJSFile)" LogicalName="_framework/$(BlazorServerJSFilename)" />
+      <EmbeddedResource Include="$(BlazorServerJSFile)" LogicalName="_framework/$(BlazorServerJSFilename)" Condition="'$(BuildNodeJS)' != 'false' and '$(BuildingInsideVisualStudio)' != 'true'"/>
```

Comment:
Why does this line and https://github.com/dotnet/aspnetcore/pull/45883/files#diff-6dab0588e76e66a0557fb78a2795bf8fe178f13f93ec7818d378f0ee7987ac34R77 need to be conditional❔

Reply:
Because this is the file we're deleting, it won't exist unless you build nodeJs. Maybe it'd be better to condition on the file existing, in case a dev follows a NodeJs build with a NoBuildNodeJs build.